### PR TITLE
Validate Duration.toObject() argument

### PR DESCRIPTION
**Pull request by vlad-zhukov**
Check that the argument is an object and has any units, otherwise create an invalid Duration.

Related to #186.

### CODE REVIEW

**Comment by vlad-zhukov** [comment]
As I expected lots of tests failed but CI is green. The setup is clearly not working right.

**Comment by icambron** [comment]
Arg, I've had this problem before: the tests are failing but the CI isn't reporting it. I'll see if I can sort that out.

**Comment by icambron** [comment]
OK, if you rebase onto master, the CI automation should now report failure.

**Comment by vlad-zhukov** [comment]
Rebased. So what do you think about checks like this? Should such methods return an invalid instance or simply throw an early error?

File: src/duration.js [hunk 1]
```diff
@@ -197,8 +198,26 @@ export class Duration {
    * @return {Duration}
    */
   static fromObject(obj) {
+    if (obj == null || typeof obj !== 'object') {
+      return Duration.invalid(INVALID_INPUT);
```
Comment (icambron) [inline src/duration.js]: This should throw. This is much more likely a programmer error than a data problem. It's similar to how we throw when the user specifies invalid units.

File: src/duration.js [hunk 2]
```diff
@@ -197,8 +198,26 @@ export class Duration {
...
+
+    let hasUnits = false;
+    for (let i = 0, l = orderedUnits.length; i < l; i++) {
+      if (values[orderedUnits[i]]) {
+        hasUnits = true;
+        break;
+      }
+    }
+
+    if (hasUnits === false) {
```
Comment (icambron) [inline src/duration.js]: flip the check:

```js
if (hasUnits) {
 return new Duration({
   values: 
   // ...
} else
  throw ...
}

File: src/duration.js [hunk 3]
```diff
@@ -197,8 +198,26 @@ export class Duration {
...
+      return Duration.invalid(INVALID_INPUT);
```
Comment (icambron) [inline src/duration.js]: I think this is throwable too

File: src/duration.js [hunk 4]
```diff
@@ -197,8 +198,26 @@ export class Duration {
...
   static fromObject(obj) {
+    if (obj == null || typeof obj !== 'object') {
+      return Duration.invalid(INVALID_INPUT);
+    }
+
+    const values = Util.normalizeObject(obj, Duration.normalizeUnit, true);
+
+    let hasUnits = false;
+    for (let i = 0, l = orderedUnits.length; i < l; i++) {
+      if (values[orderedUnits[i]]) {
```
Comment (icambron) [inline src/duration.js]: I think we can just check if `values` has any `ownProperties`, right? `Object.keys(values).length === 0`

**Comment by vlad-zhukov** [comment]
Thanks for comments, fixed. I am going to go through test errors soon and see what causes them.

**Comment by icambron** [comment]
@vlad-zhukov just a reminder that this is still here

**Comment by vlad-zhukov** [comment]
Hey! I'm not going to finish this because I don't use luxon anymore.

**Comment by icambron** [comment]
okie-dokie

**Pull request closed without merging**